filters.go

```diff
@@ -9,6 +9,15 @@ type Filter struct {
 	Field        *Filter             `json:"field,omitempty"`
 	Fields       []*Filter           `json:"fields,omitempty"`
 	ExtractionFn *FilterExtractionFn `json:"extractionFn,omitempty"`
+	QueryFilter  *Query              `json:"query,omitempty"`
+	Values       []string            `json:"values,omitempty"`
+}
+
+// QueryFilter is used for search filter
+// https://druid.apache.org/docs/latest/querying/filters.html
+type QueryFilter struct {
```

Comment:
how is this different from the base Filter struct?

Reply:
the filter query needs to have `type` and `value` inside `query` field..
```
{
    "filter": {
        "type": "search",
        "dimension": "product",
        "query": {
          "type": "insensitive_contains",
          "value": "foo"
        }
    }
}
```

I didn't want to use the same struct `Filter` in case it has other fields. I checked the documentation again, it has a field for `CaseSensitive` which I added just now.